src/App.jsx

```diff
@@ -1,58 +1,107 @@
+const SORT_FIELD_LENGTH = 'length';
+
 export const goodsFromServer = [
   'Dumplings',
   'Carrot',
   'Eggs',
-  'Ice cream',
   'Apple',
   'Bread',
   'Fish',
   'Honey',
   'Jam',
+  'Ice cream',
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function getPreparedGoods(goods, { sortField, reverseField }) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((element1, element2) => {
+      switch (sortField) {
+        case SORT_FIELD_LENGTH:
+          return element1[sortField] - element2[sortField];
+
+        case SORT_FIELD_ALPHABET:
+          return element1.localeCompare(element2);
+
+        default:
+          return 0;
+      }
+    });
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  if (reverseField) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseField, setReverseField] = useState('');
+  const visibleGoods = getPreparedGoods(goodsFromServer,
+    { sortField, reverseField });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(SORT_FIELD_ALPHABET)}
+          type="button"
+          className={cn('button is-info',
+            { 'is-light': sortField !== SORT_FIELD_ALPHABET })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+          type="button"
+          className={cn('button is-success',
+            { 'is-light': sortField !== SORT_FIELD_LENGTH })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setReverseField(!reverseField)}
+          type="button"
+          className={cn('button is-warning',
+            { 'is-light': !reverseField })}
+        >
+          Reverse
+        </button>
+
+        {
+          (sortField || reverseField) && (
+            <button
+              onClick={() => {
+                setReverseField('');
+                setSortField('');
+              }}
```

Comment:
same. make handler in component and pass reference here
```
const handleReset = ) => {
    setIsReversed(false);
    setSortField(SORT_FIELD.NONE);
 }
```
```suggestion
              onClick={handleReset}
```